Sources/Requests/TextDocumentSourceKitOptionsHandler.swift

```diff
@@ -246,28 +230,26 @@ final class TextDocumentSourceKitOptionsHandler {
         var index = 0
         let count = rawArguments.count
 
+        let indexingObjCHeader = language == .objective_c && contentToQuery.hasSuffix(".m")
+
+        if indexingObjCHeader {
+            // Xcode index builds adds these arguments to the beginning.
+            compilerArguments.append("-x")
+            compilerArguments.append("objective-c")
+        }
+
+        // Filtering/processing arguments that are either not needed for indexing or cause problems with it
         while index < count {
             let arg = rawArguments[index]
 
-            // Skip swiftc executable and wrapper arguments
-            if arg.contains("-Xwrapped-swift") || arg.hasSuffix("worker") || arg.hasPrefix("swiftc") {
```

Comment:
Logic above already skips swiftc, so no need to check it here. I think we might not need the worker one too (but I never bumped into it so feel free to correct me)